mypy/errors.py

```diff
@@ -262,7 +262,7 @@ def report(self,
             line: line number of error
             column: column number of error
             message: message to report
-            code: error code (defaults to 'misc' for 'error' severity)
+            code: error code (defaults to 'misc'; not shown for notes)
```

Comment:
This is a very good choice! Could you please add test (for example with protocols, they generate long notes) checking that codes are not shown on notes? Maybe also mention this in docs.